miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java

```diff
@@ -99,6 +118,134 @@ protected BoxDto asDto(Box model) {
+   */
+  private List<List<String>> getBoxContents(Box box) throws IOException {
+    List<List<String>> boxContents = new ArrayList<>();
+
+    // Use TreeMap to iterate through the contents of the box in order of box position
+    for (Map.Entry<String, BoxableView> entry : new TreeMap<>(box.getBoxables()).entrySet()) {
+      String position = entry.getKey();
+      BoxableView boxableView = entry.getValue();
+
+      String name = boxableView.getName();
+      String alias = boxableView.getAlias();
+      String barcode = boxableView.getIdentificationBarcode();
+
+      if (detailedSample) {
+        String externalName = findExternalName(boxableView);
+        String numSlides = findNumSlides(boxableView);
+
+        boxContents.add(Arrays.asList(position, name, alias, barcode, externalName, numSlides));
+      } else {
+        boxContents.add(Arrays.asList(position, name, alias, barcode));
+      }
+    }
+
+    return boxContents;
+  }
+
+  private String findExternalName(BoxableView boxableView) throws IOException {
+    DetailedSample detailedSample;
+
+    switch(boxableView.getId().getTargetType()) {
+    case SAMPLE:
+      detailedSample = (DetailedSample) extractSample(boxableView);
+      break;
+    case LIBRARY:
+      detailedSample = (DetailedSample) extractLibrary(boxableView).getSample();
+      break;
+    case DILUTION:
+      detailedSample = (DetailedSample) extractDilution(boxableView).getLibrary().getSample();
+      break;
+    default: // Can't find external name for a Pool or otherwise
+      return "n/a";
+    }
+
+    return extractSampleIdentity(detailedSample).getExternalName();
+  }
+
+  private SampleIdentity extractSampleIdentity(DetailedSample detailedSample) {
+    while (!isIdentitySample(detailedSample)) {
+      if (detailedSample == null) {
+        throw new IllegalStateException("No identity found in hierarchy");
+      }
+
+      detailedSample = detailedSample.getParent();
+    }
+
+    // Load all fields from Hibernate object to enable cast to SampleIdentity
+    return (SampleIdentity) deproxify(detailedSample);
+  }
+
+  private Sample extractSample(BoxableView boxableView) throws IOException {
+    return sampleService.get(boxableView.getId().getTargetId());
+  }
+
+  private Library extractLibrary(BoxableView boxableView) throws IOException {
+    return libraryService.get(boxableView.getId().getTargetId());
+  }
+
+  private LibraryDilution extractDilution(BoxableView boxableView) throws IOException {
+    return libraryDilutionService.get(boxableView.getId().getTargetId());
+  }
+
+  private String findNumSlides(BoxableView boxableView) throws IOException {
+    if (isSample(boxableView)) {
+      DetailedSample sample = (DetailedSample) extractSample(boxableView);
+
+      if (isSampleSlide(sample)) {
+        return Integer.toString(((SampleSlide) sample).getSlides());
+      }
+    }
+
+    return "n/a";
+  }
+
+
+  private boolean isSampleSlide(DetailedSample sample) {
```

Comment:
Add to `LimsUtils` instead, making it similar to the other similar methods. e.g. `isTissueSample`